pytorch_lightning/trainer/supporters.py

```diff
@@ -457,6 +464,19 @@ def _wrap_loaders_max_size_cycle(self) -> Any:
             )
             state.reset()
 
+    def _apply_cycle_iterator_length(self):
```

Comment:
Note for reviewing: I am not sure happy with this function, but unfortunately, the length computation requires to be computed only once all the dataloaders have been modified.
Alternatives: use `self.length / world_size` within the CycleIterator, but this could lead to incorrect results if the users had custom dataloaders, samplers.

Reply:
this info could be added as a comment or docstring to this method.

Reply:
Done !